src/index.js

```diff
@@ -77,7 +77,8 @@ async function run(request, context) {
   params.GOOGLE_PRIVATE_KEY = context.env.GOOGLE_PRIVATE_KEY;
   params.GOOGLE_PROJECT_ID = context.env.GOOGLE_PROJECT_ID;
 
-  return runExec(params, pathname.split('/').pop(), context.log);
+  // take final URL path segment and then substitute @ with / to allow for /queries subfolders
+  return runExec(params, pathname.split('/').pop().replaceAll('@', '/'), context.log);
```

Comment:
The issue is that `.pop()` takes only the last path segment (the file) and we lose the folder name to look it up later at https://github.com/adobe/helix-run-query/blob/main/src/util.js#L22.  And taking the last two segments is invalid since not all have a subfolder under queries.  And removing the first N path segments is invalid because sometimes `/helix-services/run-query@ci####` (2 segments) is used while other times `/helix-services/run-query/ci####` (3 segments) is used.  After several iterations, it became a question of finding an approach that would always work.

Reply:
Look for the `@` then split at `/` and drop the first element.

Reply:
That will work for most but not all patterns (see the 3 segment example above).  But your point is clear and this is a solvable problem.  Use / as the path separator.  I will close this PR without merging and open more atomic ones for each feature/query.